### PR TITLE
Add APC e-dictionary to dictionaries (原住民族委員會 - 原住民族語言線上詞典)

### PR DESCRIPTION
**Pull request by mlouielu**
provider: 原住民族委員會 - 原住民族語言線上詞典
site: https://e-dictionary.apc.gov.tw/

### CODE REVIEW

**Comment by mlouielu** [comment]
Search result of "utux"
Selection_01201

Search result of "tux"

If exact matching or fuzzy matching didn't have result, it will return an indigenous word of "nothing" with "result not found"

Selection_01202

File: zdict/dictionaries/apc.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
+import random
+import json
+import requests
+from collections import defaultdict
```
Comment (iblis17) [inline zdict/dictionaries/apc.py:4]:
```suggestion

from collections import defaultdict
```

File: zdict/dictionaries/apc.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
+import random
```
Comment (iblis17) [inline zdict/dictionaries/apc.py:1]: please sort these `import` statement in alphabet order

File: zdict/dictionaries/apc.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+from bs4 import BeautifulSoup
+
+from zdict.dictionary import DictBase
+from zdict.exceptions import NotFoundError
+from zdict.models import Record
+
+
+class ApcDict(DictBase):
+    BASE_URL = "https://e-dictionary.apc.gov.tw"
+    POST_API = "https://e-dictionary.apc.gov.tw/index.aspx/_crossTribes"
```
Comment (iblis17) [inline zdict/dictionaries/apc.py:15]:
```suggestion
    POST_API = "{}/index.aspx/_crossTribes".format(BASE_URL)
```

File: zdict/dictionaries/apc.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    API = "https://e-dictionary.apc.gov.tw/search/list.htm"
```
Comment (iblis17) [inline zdict/dictionaries/apc.py:16]:
```suggestion
    API = "{}/search/list.htm".format(BASE_URL)
```

File: zdict/dictionaries/apc.py (hunk 3 above)
Comment (mlouielu) [reply zdict/dictionaries/apc.py]: Updated

File: zdict/dictionaries/apc.py (hunk 2 above)
Comment (mlouielu) [reply zdict/dictionaries/apc.py]: Updated

**Comment by iblis17** [comment]
The linter was complained about the style

```
_________________________________ FLAKE8-check _________________________________

/Users/travis/build/zdict/zdict/zdict/dictionaries/apc.py:76:80: E501 line too long (82 > 79 characters)

/Users/travis/build/zdict/zdict/zdict/dictionaries/apc.py:86:80: E501 line too long (87 > 79 characters)

/Users/travis/build/zdict/zdict/zdict/dictionaries/apc.py:99:80: E501 line too long (83 > 79 characters)

/Users/travis/build/zdict/zdict/zdict/dictionaries/apc.py:104:80: E501 line too long (87 > 79 characters)

/Users/travis/build/zdict/zdict/zdict/dictionaries/apc.py:110:80: E501 line too long (80 > 79 characters)

/Users/travis/build/zdict/zdict/zdict/dictionaries/apc.py:137:80: E501 line too long (83 > 79 characters)

/Users/travis/build/zdict/zdict/zdict/dictionaries/apc.py:172:80: E501 line too long (82 > 79 characters)

```

File: zdict/dictionaries/apc.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+from zdict.dictionary import DictBase
+from zdict.exceptions import NotFoundError
+from zdict.models import Record
+
+
+class ApcDict(DictBase):
+    BASE_URL = "https://e-dictionary.apc.gov.tw"
+    POST_API = "{}/index.aspx/_crossTribes".format(BASE_URL)
+    API = "{}/search/list.htm".format(BASE_URL)
```
Comment (M157q) [inline zdict/dictionaries/apc.py:17]: `POST_API` should be `API`.

File: zdict/dictionaries/apc.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+    # Ref:
```
Comment (M157q) [inline zdict/dictionaries/apc.py:19]: Why we need a `NOT_FOUND_WORD_INDIGENOUS` here?
I don't see the need for this.

File: zdict/dictionaries/apc.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+                self.color.print("{i:-2d}. {d}".format(i=i + 1, d=d), indent=4)
+
+            self.color.print("detail: ", "indigo", end="", indent=4)
+            self.color.print(content["exact_sources"][source]["link"], "white")
+            print()
+        if not content["exact_sources"]:
+            print()
+            self.color.print(
+                random.choice(self.NOT_FOUND_WORD_INDIGENOUS), "red", indent=4, end="",
+            )
```
Comment (M157q) [inline zdict/dictionaries/apc.py:87]: I think we can just show the message "無精確搜尋結果" when content['exact_sources'] is empty?

File: zdict/dictionaries/apc.py [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+            for word in content["fuzzy_sources"][source]:
+                self.color.print(word["title"], "lindigo", indent=5)
+                for i, d in enumerate(word["defs"]):
+                    self.color.print("{i:-2d}. {d}".format(i=i + 1, d=d), indent=6)
+            print()
+        if not content["fuzzy_sources"]:
+            print()
+            self.color.print(
+                random.choice(self.NOT_FOUND_WORD_INDIGENOUS), "red", indent=4, end="",
+            )
```
Comment (M157q) [inline zdict/dictionaries/apc.py:105]: I think we can just show "無模糊搜尋結果" when content['fuzzy_sources'] is empty?

File: zdict/dictionaries/apc.py [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+        "ungat",  # 太魯閣語
+        "nai'",  # 撒奇萊雅語
+        "ungac",  # 都達語 (賽德克語)
+        "uka",  # 都達語 (賽德克語)
+        "ini",  # 德固達雅語 (賽德克語)
+        "ungac",  # 德路固語 (賽德克語)
+        "ungat",  # 德路固語 (賽德克語)
+        "uka'a",  # 拉阿魯哇語
+        "'akia",  # 卡那卡那富語
+    ]
```
Comment (M157q) [inline zdict/dictionaries/apc.py:53]: Why we need a `NOT_FOUND_WORD_INDIGENOUS` here?
I don't see the need for this.

**Pull request merged**